src/main/java/com/redislabs/PropertyData.java

```diff
@@ -43,7 +43,7 @@ public Object convertToObject(String val) {
       // try first to parse as timestamp
       try {
         long timestamp = Long.parseLong(val);
-        return new Date(timestamp);
+        return new Timestamp(timestamp);
```

Comment:
Please explain this change

Reply:
Not relevant after the latest change of doing the retry in-place.